### PR TITLE
Added HTTP/2 Support

### PR DESCRIPTION
**Pull request by dr4g0n369**
I made slight changes in your code in order to implement HTTP/2, please let me know if these are no problem:
- Changed `hc` to `c.ClientInterface`
- Removed `hc` argument in function `trunner.RunApiTests` 

Here is a screenshot of the test I ran to check the request was HTTP/2
Screenshot (3)

Please let me know if any changes are required
#2

### CODE REVIEW

File: cmd/offat/main.go [hunk 1]
```diff
@@ -112,12 +117,21 @@ func main() {
...
+		httpCfg := http.NewConfig(config.RequestsPerSecond, config.SkipTlsVerification, config.Proxy)
+		new_http := http.NewHttp(httpCfg)
+		hc = new_http.Client.FHClient
+	}
 
 	// Test server connectivity
-	url := *parser.Doc.GetBaseUrl()
-	resp, err := hc.Client.FHClient.Do(url, fasthttp.MethodGet, nil, nil, nil)
+	// url := *parser.Doc.GetBaseUrl()
+	url := "https://petstore.swagger.io/v2/store/inventory" // Uncomment for testing
```
Comment (dmdhrumilmistry) [inline cmd/offat/main.go]: we can remove hardcoded values
```suggestion
	url := *parser.Doc.GetBaseUrl()
```

File: cmd/offat/main.go [hunk 2]
```diff
@@ -112,12 +117,21 @@ func main() {
...
+
+	// resp, err := hc.Client.FHClient.Do(url, fasthttp.MethodGet, nil, nil, nil)
+	resp, err := hc.Do(url, fasthttp.MethodGet, nil, nil, nil)
```
Comment (dmdhrumilmistry) [inline cmd/offat/main.go]: can we remove hard coded values?

File: cmd/offat/main.go [hunk 3]
```diff
@@ -103,6 +106,8 @@ func main() {
 		log.Error().Err(err).Msg("failed to set baseUrl")
 	}
 
+	fmt.Println(config.BaseUrl)
```
Comment (dmdhrumilmistry) [inline cmd/offat/main.go]: we can remove debugging statements

File: cmd/offat/main.go [hunk 4]
```diff
@@ -112,12 +117,21 @@ func main() {
...
 	// Create http client and config
-	httpCfg := http.NewConfig(config.RequestsPerSecond, config.SkipTlsVerification, config.Proxy)
-	hc := http.NewHttp(httpCfg)
+	var hc c.ClientInterface // changed hc to interface
+	if config.HTTP2 != nil && *config.HTTP2 {
+		hc = http.NewConfigHttp2(config.RequestsPerSecond, config.SkipTlsVerification, config.Proxy)
+	} else {
+		httpCfg := http.NewConfig(config.RequestsPerSecond, config.SkipTlsVerification, config.Proxy)
+		new_http := http.NewHttp(httpCfg)
+		hc = new_http.Client.FHClient
```
Comment (dmdhrumilmistry) [inline cmd/offat/main.go]: When will this be used? Most of the vars seems to be hardcoded below

File: pkg/http/http2.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+type ThrottledTransport struct {
+	roundTripperWrap http.RoundTripper
+	ratelimiter      *rate.Limiter
+}
+
+type CustomClient struct {
+	Client *http.Client
+}
+
```
Comment (dmdhrumilmistry) [inline pkg/http/http2.go]: can we have tests for http2 client?

File: cmd/offat/main.go (hunk 3 above)
Comment (dr4g0n369) [reply cmd/offat/main.go]: Sorry for that. I will make sure to get rid of the debugging statements.

File: pkg/http/http2.go (hunk 5 above)
Comment (dr4g0n369) [reply pkg/http/http2.go]: Yup surely. I will write the tests.

File: cmd/offat/main.go (hunk 4 above)
Comment (dr4g0n369) [reply cmd/offat/main.go]: Alright, I will remove the unnecessary `new_http` variable.

**Comment by dr4g0n369** [comment]
I have made the changes. Please check them to see if they are fine.

Thank you

**Comment by dmdhrumilmistry** [comment]
> I have made the changes. Please check them to see if they are fine.
> 
> Thank you

Thanks for contributing to the project, I'll re-review it once again whenever I get time.

File: pkg/http/http_test.go [hunk 6]
```diff
@@ -33,3 +33,25 @@ func TestHttpClient(t *testing.T) {
...
+		hc.Responses = c.MakeConcurrentRequests(hc.Requests, hc)
+
+		for _, connResp := range hc.Responses {
+			if connResp.Error != nil {
+				t.Fatalf("failed to make concurrent request: %v\n", connResp.Error)
+			}
+		}
+	})
+
+}
```
Comment (dmdhrumilmistry) [inline pkg/http/http_test.go]: can we move this test to `http2_test.go`

**Review by dmdhrumilmistry — CHANGES_REQUESTED** [review]
can you make necessary changes? I'll approve and merge the PR

File: pkg/http/http_test.go (hunk 6 above)
Comment (dr4g0n369) [reply pkg/http/http_test.go]: I have made the changes.

**Review by dmdhrumilmistry — APPROVED** [review]

**Pull request merged**